File: sourcecode/src/DoublyLinkedList/LinkedList.java
package List;
public class LinkedList {
	private String value;
	private LinkedList next;
	
	public LinkedList(String value) {		
		this.value = value;
		this.next = null;
	}
	public void insertEnd( String value){
		LinkedList temp = this;
		while (temp.next != null)
			temp = temp.next;
		temp.next = new LinkedList(value);
		
		
	}
	public void deleteEnd(){
		
	}
	public void display(LinkedList p){
		while ( p != null) {
			System.out.print(p.value+"----> ");
			p= p.next;
		}
		
	}
	public LinkedList recReverse(LinkedList p) {
		if (p == null || p.next == null)
			return p;
		LinkedList temp = p.next;
		p.next = null;
		LinkedList count = recReverse(temp);
		temp.next = p;
		
		return count;	
		
	}

	public static void main(String[] args) {
		LinkedList l = new LinkedList("1");
		l.insertEnd("2");	
		l.insertEnd("3");	
		l.insertEnd("4");	
		l.display(l);
		System.out.println();
		l = l.recReverse(l);
		l.display(l);
		
	}

}

File: C++datastructure/LinkedList.cpp
// LinkedList.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <iostream>
using namespace std;
class LinkedList {
private:
	struct node {
		int data;
		node *next;
		};
	node *p;
public:
	LinkedList(){
		p = NULL;

		}
	~LinkedList(){
		if (p == NULL)
			return;
		node *temp = p;
		while (temp != NULL) {
			p = temp;
			temp = temp->next;
			delete p;

		}

	}
	void lastToFirst(){
		if (p == NULL)
			return;
		node *temp = p;
		node *last = NULL;
		while(temp->next != NULL){
			last = temp;
			temp = temp->next;

		}
		last->next = NULL;
		temp->next = p;
		p = temp;
	}
	};
	LinkedList(int num) {
		p = new node;
		p->data = num;
		p->next = NULL;
		}
	void deleteAlternate(){
		node *temp = p;
		while (true) {
			node *temp1 = temp->next;
			if (temp1 == NULL)
				break;
			node *temp2 = temp1->next;
			if (temp2 == NULL) {
				temp->next = NULL;
				cout << "Just before deleting temp next";
				delete temp->next;
				break;
			}
			temp->next = temp2;
			temp1->next = NULL;
			delete temp1->next;
			temp = temp2;

		}

	}
	void deleteEnd() {
		if (p == NULL)
			return;
		node *temp = p;
		node *last = NULL;
		while (temp->next != NULL){
			last = temp;
			temp = temp->next;

		}
		delete temp;
		last->next = NULL;
	}
	void append(int num){
		if (p == NULL){
			p = new node;
			p->data = num;
			p->next = NULL;
			return;
			}
		node *temp = p;
		while (temp->next != NULL){
			temp = temp->next;
			}
		temp->next = new node;
		temp->next->data = num;
		temp->next->next = NULL;
		}
	void display(){
		node *temp = p;
		while (temp != NULL){
			cout << temp->data << endl;
			temp = temp->next;
			}
		}
	};


int _tmain(int argc, _TCHAR* argv[])
{
LinkedList l;
l.append(1);
l.append(2);
l.append(3);
l.append(7);
cout << endl;
l.display();


	return 0;
}

